This patch does not look safe and it has not been accepted upstream.
Index: t1lib-5.1.1/lib/t1lib/t1base.c
===================================================================
--- t1lib-5.1.1.orig/lib/t1lib/t1base.c	2005-05-18 05:08:01.000000000 +0200
+++ t1lib-5.1.1/lib/t1lib/t1base.c	2008-01-06 20:48:32.000000000 +0100
@@ -455,7 +455,7 @@
 	   to test_for_t1_file() filename is substituted by an emty
 	   string if the file was not found: */
 	sprintf( err_warn_msg_buf, "Type 1 Font file %s.[pfa/pfb] not found (FontID=%d, SearchPath=%s)",
-		 linebuf, m-1, T1_GetFileSearchPath(T1_PFAB_PATH));
+		 linebuf, m-1, intT1_GetFileSearchPath(T1_PFAB_PATH));
 	if ((test_for_t1_file( &linebuf[0]))){
 	  T1_PrintLog( "intT1_scanFontDBase()", err_warn_msg_buf, T1LOG_WARNING);
 	  located=0;
@@ -579,7 +579,7 @@
 	   to test_for_t1_file() filename is substituted by an emty
 	   string if the file was not found: */
 	sprintf( err_warn_msg_buf, "Type 1 Font file %s.[pfa/pfb] not found (FontID=%d, SearchPath=%s)",
-		 linebuf, m-1, T1_GetFileSearchPath(T1_PFAB_PATH));
+		 linebuf, m-1, intT1_GetFileSearchPath(T1_PFAB_PATH));
 	if ((test_for_t1_file( &linebuf[0]))){
 	  T1_PrintLog( "intT1_scanFontDBase()", err_warn_msg_buf, T1LOG_WARNING);
 	  located=0;
Index: t1lib-5.1.1/lib/t1lib/t1env.c
===================================================================
--- t1lib-5.1.1.orig/lib/t1lib/t1env.c	2008-01-06 20:48:32.000000000 +0100
+++ t1lib-5.1.1/lib/t1lib/t1env.c	2008-01-06 20:48:32.000000000 +0100
@@ -807,6 +807,14 @@
 
 }
 
+/* intT1_GetFileSearchPath(): 
+ call T1_GetFileSearchPath and truncate the line to avoid overflowing
+ error buffers*/ 
+char *intT1_GetFileSearchPath( int type)
+{
+   char  *res = T1_GetFileSearchPath(type);
+   res[900] = 0;
+}
 
 /* T1_AddToFileSearchPath(): Add the specified path element to
    the specified search path. If the existing path is the default path,
Index: t1lib-5.1.1/lib/t1lib/t1env.h
===================================================================
--- t1lib-5.1.1.orig/lib/t1lib/t1env.h	2005-05-16 20:37:43.000000000 +0200
+++ t1lib-5.1.1/lib/t1lib/t1env.h	2008-01-06 20:48:32.000000000 +0100
@@ -35,6 +35,7 @@
 int T1_SetFileSearchPath( int type, char *pathname);
 int T1_AddToFileSearchPath( int pathtype, int mode, char *pathname);
 char *T1_GetFileSearchPath( int type);
+char *intT1_GetFileSearchPath( int type);
 int T1_SetFontDataBase( char *filename);
 int T1_AddFontDataBase( int mode, char *filename);
 int T1_SetFontDataBaseXLFD( char *filename);
@@ -49,6 +50,7 @@
 extern int T1_SetFileSearchPath( int type, char *pathname);
 extern int T1_AddToFileSearchPath( int pathtype, int mode, char *pathname);
 extern char *T1_GetFileSearchPath( int type);
+extern char *intT1_GetFileSearchPath( int type);
 extern int T1_SetFontDataBase( char *filename);
 extern int T1_AddFontDataBase( int mode, char *filename);
 extern int T1_SetFontDataBaseXLFD( char *filename);
